### DISCUSSION TITLE
Support for search refinement

### DISCUSSION THREAD

**Discussion by lroos** [pnp/pnpcore#923] [Q&A] [answered]
I see there's support for returning refiners: [Getting Refiners for the Returned Search Results](https://pnp.github.io/pnpcore/using-the-sdk/search-intro.html#getting-refiners-for-the-returned-search-results), but no implementation of [Refinement Filters](https://docs.microsoft.com/en-us/sharepoint/dev/general-development/sharepoint-search-rest-api-overview#refinementfilters), so there's no way to pass these into the query. See also [Query Refinement](https://docs.microsoft.com/en-us/sharepoint/dev/general-development/query-refinement-in-sharepoint) for example usage.  The [SearchAsync](https://github.com/pnp/pnpcore/blob/5af511b6a028ea7bdd009e3617771bab1c13ce75/src/sdk/PnP.Core/Model/SharePoint/Core/Internal/Web.cs#L1308) only supports a GET request, which limits longer filters or advanced options. 

Would the workaround be to send a raw request using ExecuteRequestAsync() and what would that look like? Or are you open to a PR to add support for refinementFilters param? If we were to also add support for sending the request as a POST, what would the API surface look like for SearchAsync? Assuming there could be an overload, or adding the method to SearchOptions.

**Comment by jansenbe** [comment, marked as answer]
@lroos : thanks for using PnP Core SDK and providing the feedback. I'm open for a PR that provides refinement filter capabilities, there's nothing wrong with updating our internal implementation to use a POST instead of a GET to ensure we're not hitting URL length limitations.

If you prefer to only solve this for your solution you can indeed do this via a custom API call: see https://pnp.github.io/pnpcore/using-the-sdk/basics-customapirequests.html#making-a-custom-spo-rest-request for details on how to so.

**Reply by jansenbe** [reply]
Implemented by @lroos (#924). Thanks!!
